webextension/privileged/ui/CloudDownloadsView.jsm

```diff
@@ -0,0 +1,739 @@
+    menuItem.setAttribute("image", this._iconURL(providerDetail.displayName));
+    return menuItem;
+  },
+
+  _setMoveDownloadMenuPopUpAttributes(menuItem) {
+    menuItem.removeAttribute("hidden");
+    const menuPopup = menuItem.menupopup;
+    menuPopup.removeAttribute("hidden");
+    return menuPopup.firstChild;
+  },
+
+  _displayMoveToCloudContextMenuItem(downloadElement) {
+    const aPopupMenu = downloadElement.ownerDocument.getElementById("downloadsContextMenu");
+    let menuItem = aPopupMenu.getElementsByAttribute("id", "moveDownload")[0];
+    if (menuItem) {
+      menuItem.setAttribute("hidden", "true");
+
+      // Exit if user has opted to store subsequent downloads in cloud provider folder
+      if (CloudDownloadsInternal.preferredProviderKey) {
+        return;
+      }
+
+      if (CloudDownloadsInternal.checkIfExistingCloudProviderDownloadSettings()) {
+        return;
+      }
+
+      const downloadType = downloadElement.getAttribute("cloudstorage");
+      // Downloaded Item is saved in local downloads folder
+      if (this.providers.size > 1) {
+        let subMenuItem =  this._setMoveDownloadMenuPopUpAttributes(menuItem);
+        this.providers.forEach((value, key) => {
+          if (this._formatProviderName(value.displayName) !== downloadType) {
+            subMenuItem = this._setMenuItemAttributes(subMenuItem, key, value, downloadElement);
+            subMenuItem = subMenuItem.nextSibling;
+          }
+        });
+        if (downloadType != "local") {
+          subMenuItem = this._setMenuItemAttributes(subMenuItem, "local", {displayName: "Local Downloads"}, downloadElement);
+        }
+      } else {
+        const provider = downloadType != "local" ?
+          ["local", {displayName: "Local Downloads"}] : this.providers.entries().next().value;
+        menuItem = this._setMenuItemAttributes(menuItem, provider[0], provider[1], downloadElement);
+        menuItem.setAttribute("label", "Move To " + provider[1].displayName);
+        menuItem.removeAttribute("hidden");
+      }
+    }
+  },
+
+  handleEvent(event) {
+    // Handle multiple provider displayed as options in notification
+    if (event.target.parentElement.id === "multiProviderSelect") {
+      const cloudDownloadSave = event.currentTarget.querySelector("#cloudDownloadSave");
+      cloudDownloadSave.setAttribute("label", event.target.label);
+      cloudDownloadSave.setAttribute("providerKey", event.target.id);
+      cloudDownloadSave.removeAttribute("disabled");
+      return;
+    }
+
+    // Handle rendering right provider in context menu when shown
+    if (event.type === "popupshowing") {
+      if (event.target.triggerNode.parentNode.id !== "downloadsListBox") {
+        const menuItem = event.target.getElementsByAttribute("id", "moveDownload")[0];
+        if (menuItem) {
+          menuItem.setAttribute("hidden", "true");
+        }
+        return;
+      }
+
+      const element = event.target.triggerNode;
```

Comment:
You're checking whether `event.target.triggerNode` is null, but a bit earlier you use it without checking if it's null. Either the nullcheck is superfluous or it should come before using `event.target.triggerNode`

Reply:
done